framework/application/android_window.cpp

```diff
@@ -158,13 +159,25 @@ decode::Window* AndroidWindowFactory::Create(
     GFXRECON_UNREFERENCED_PARAMETER(height);
     GFXRECON_UNREFERENCED_PARAMETER(force_windowed);
 
-    return android_context_->GetWindow();
+    tmp_window = nullptr;
+    android_context_->requestNativeWindow(width, height);
+    AndroidWindow* tmpwin = nullptr;
+    if (tmp_window != nullptr)
+    {
+        tmpwin = new AndroidWindow(android_context_, tmp_window);
+        GFXRECON_LOG_INFO("Got android window %p\n", tmp_window);
+    }
+    else
+    {
+        GFXRECON_LOG_WARNING("Get android window failed");
+    }
+    return tmpwin;
 }
 
 void AndroidWindowFactory::Destroy(decode::Window* window)
 {
-    // Android currently has a single window whose lifetime is managed by AndroidContext.
-    GFXRECON_UNREFERENCED_PARAMETER(window);
+    int32_t windowidx = created_window_.at(window) - 1;
```

Comment:
If you increment `created_window_count_` after inserting a new entry in the map, you can also get rid of this `- 1` here.